File: 3639-zero-array-transformation-i/zero-array-transformation-i.go
func isZeroArray(nums []int, queries [][]int) bool {
    n := len(nums);
    count := make([]int, n + 1);

    for _, q := range queries {
        count[q[0]]++;
        count[q[1] + 1]--;
    }

    sum := 0;
    for i := 0; i < n; i++ {
        sum += count[i];
        count[i] = sum;
        if count[i] < nums[i] {
            return false;
        }
    }

    return true;
}

File: 3639-zero-array-transformation-i/zero-array-transformation-i.java
class Solution {
    public boolean isZeroArray(int[] nums, int[][] queries) {
        int n = nums.length;
        int[] count = new int[n + 1];

        for (int[] q : queries) {
            int l = q[0], r = q[1];
            count[l] += 1;
            if (r + 1 < n) {
                count[r + 1] -= 1;
            }
        }

        for (int i = 1; i < n; i++) {
            count[i] += count[i - 1];
        }

        for (int i = 0; i < n; i++) {
            if (nums[i] > count[i]) {
                return false;
            }
        }

        return true;
    }
}